### ISSUE TITLE
Question: Getting NPE when running dspot

### ISSUE THREAD

**Issue by borisbaldassari** [STAMP-project/dspot#741]
#### Characteristics
- **Issue Type**: question/bug
- **Reproducibility**: always
- **Severity**: major
- **Tool/Service/Component**: Dspot
- **Execution Environment**: Linux Debian 9, amd64, JDK Oracle 1.8.0_201
- **Reporter**: see github login

#### Description

I want to apply dspot to dspot. So I created the following properties file: 
```java
# Relative path to the project root.
project=..
# Path to the current module where we want to execute DSpot, relative to the project root
targetModule=dspot/
# Relative path to the source project from this properties file
src=src/main/java/
# Relative path to the test source project from this properties file
testSrc=src/test/java
# Java version used
javaVersion=8
# (Optional) Path to the output folder, default to "output_diversify"
outputDirectory=dspot-out/
```

And executed in the dspot directory the following command:
```shell
boris@kadath:dspot$ java -jar target/dspot-2.0.1-SNAPSHOT-jar-with-dependencies.jar --path-to-properties dspot.properties
```

And it fails with an exception, see attached file. 
[dspot_exec_failed.log](https://github.com/STAMP-project/dspot/files/3042008/dspot_exec_failed.log)

After execution the jars seem to have vanished, I see no dspot.out directory but there is a NUL file with the following content:
```
Parsing --binaries /home/boris/Repositories/gh_kth_dspot/dspot/src/test/resources/test-projects/target/classes/:/home/boris/Repositories/gh_kth_dspot/dspot/src/test/resources/test-projects/target/test-classes/ --class example.TestSuiteExample
File saved to the following path: /home/boris/Repositories/gh_kth_dspot/dspot/src/test/resources/sample/target/dspot/Coverage.ser
```

**Comment by danglotb** [comment]
Hello,

In one hand, If you want to run DSpot on DSpot, you should extract the running jar from the target folder because the first thing that DSpot does is the launch the following maven goals:

```sh
mvn clean test -DskipTests dependency:build-classpath -Dmdep.outputFile=target/dspot/classpath
```

And thus, the jar that you are executing is removed. From this jar, DSpot copies some classes required to observe the state of the program.

In the other hand, the required classes should be loaded in the current JVM and accessible anyway. I don't really understand. I'll investigate.

I never tried to apply DSpot on DSpot.

> After execution the jars seem to have vanished, I see no dspot.out directory but there is a NUL file with the following content:

What do you mean by `NUL` file? The file `Coverage.ser` is a serialized object that contains information about the code coverage computed by DSpot.

**Comment by borisbaldassari** [comment]
Hey Benjamin, 
Good point for the jar vanishing. I copied it elsewhere and reran the execution. It's going a bit further but it is still failing, see attached log.

[dspot.txt](https://github.com/STAMP-project/dspot/files/3042457/dspot.txt)

Regarding the `NUL` file, what I meant was: there is a file, called 'NUL', with the mentioned content. 
Anyway it's not generated anymore so it was probably due to the jar-in-the-bad-place issue.

**Comment by danglotb** [comment]
The generation of the `NUL` file has been fixed in STAMP-project/testrunner#71

Closing this issue. Thank you.
